java/src/test/java/ai/rapids/cudf/ColumnVectorTest.java

```diff
@@ -2926,6 +2926,57 @@ void testScanProduct() {
     }
   }
 
+  @Test
+  void testScanRank() {
+    try (ColumnVector v1 = ColumnVector.fromBoxedInts(1, 4, 8, 6, 10, 9, 7, 9, 5, 0, 7, 7);
```

Comment:
We are doing a scan on v1, but v1 does not contribute to the result of the rank or dense_rank scan at all.  This is kind of masked in the C++ tests, but is very obvious here. This is also true of the group by scan.  One of the reasons I am concerned about this is because in the degenerate case, especially for scan, I may have to create an arbitrary column just so I have something to run the scan on top of.

I think it might be cleaner to take a single column and if we need multiple columns then we use a struct for it. But that too is not ideal.

Reply:
I'll look into a solution, the additional instance of this issue is the `onColumn(x)` argument when creating aggregations on the Java side.